src/hooks/useFetchMovies.js

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState, useCallback } from 'react';
+import { fetchMovies } from '../helpers/MoviesHelper';
+
+const useFetchMovies = () => {
+  const [page, setPage] = useState(1);
+  const [shouldFetch, setShouldFetch] = useState(true);
+  const [movies, setMovies] = useState([]);
+
+  const fetchMore = useCallback(() => setShouldFetch(true), []);
+
+  useEffect(() => {
+    if (!shouldFetch) {
+      return;
+    }
+
+    const fetch = async () => {
+      const res = await fetchMovies(page);
+      const newMovies = res.results;
+      setShouldFetch(false);
+      setMovies(prevMovies => [...prevMovies, ...newMovies]);
```

Comment:
Maybe the error you were getting about duplicate keys is because the api returns the same movie in two different pages for some reason so here you have it twice in the array. It's not a big thing but you can check.